frontend/script.js

```diff
@@ -0,0 +1,57 @@
+function showSideBar() {
+    const sidebar = document.querySelector('.hamburger');
+    sidebar.style.display = 'flex';
+}
+
+function hideSideBar() {
+    const sidebar = document.querySelector('.hamburger');
+    sidebar.style.display = 'none';
+}
+
+// Specify the API endpoint for user data
+const apiUrl = 'http://localhost:8080/products/';
+// Make a GET request using the Fetch API
+fetch(apiUrl)
+    .then((response) => {
+        if (!response.ok) {
+            throw new Error('Network response was not ok');
+        }
+        return response.json();
+    })
+    .then((data) => {
+        data.data.forEach((element) => {
+            if ('content' in document.createElement('template')) {
+                const trendingBody = document.querySelector('#trending');
```

Comment:
I would separate out all the logic which is currently placed in this callback into multiple helpers, which could then be called in a wrapping function, just to understand what everything does. 

For example 
`function createProductImage()`, `function createProductCard()`, `function formatPrice()` or similar (naming is always tricky 😄)

Reply:
Updated fetch.js to include helper functions